app/src/main/java/com/infomaniak/mail/ui/MainViewModel.kt

```diff
@@ -196,6 +199,23 @@ class MainViewModel @Inject constructor(
     }
     //endregion
 
+    //region Current Thread
+    val currentThreadUid = MutableLiveData<String?>(null)
+
+    val closeThreadTrigger = SingleLiveEvent<Unit>()
+
+    val downloadAttachmentsArgs = SingleLiveEvent<Triple<String, String, AttachmentType>>()
+    val newMessageArgs = SingleLiveEvent<NewMessageActivityArgs>()
+    val replyBottomSheetArgs = SingleLiveEvent<Pair<String, Boolean>>()
+    val threadActionsBottomSheetArgs = SingleLiveEvent<Triple<String, String, Boolean>>()
+    val messageActionsBottomSheetArgs = SingleLiveEvent<MessageActionsArgs>()
+    val detailedContactArgs = SingleLiveEvent<Recipient>()
```

Comment:
Why were all of these extracted actually?

Reply:
Answered here : https://github.com/Infomaniak/android-kMail/pull/1573#discussion_r1425256192